App/Syncthing/syncthing_sync_check.sh

```diff
@@ -0,0 +1,284 @@
+        for device in $devices; do
+            local device_name=$(get_device_name "$device")
+            local short_id=$(echo "$device" | cut -c1-7)
+            log_message "Device $device_name ($short_id):"
+            summary="${summary}Device $device_name ($short_id):\n"
+
+            echo "$folders" | while IFS='|' read -r folder_id folder_label; do
+                if [ "$mode" = "startup" ]; then
+                    local completion=$(calculate_folder_completion "$folder_id")
+                    [ "$completion" != "100" ] && echo "not_synced" > /tmp/sync_status
+                elif [ "$mode" = "shutdown" ]; then
+                    local completion=$(calculate_total_completion "$device" "$folder_id")
+                    [ "$completion" != "100" ] && echo "not_synced" > /tmp/sync_status
+                elif [ "$mode" = "monitor" ]; then
+                    local download_completion=$(calculate_folder_completion "$folder_id")
+                    local upload_completion=$(calculate_total_completion "$device" "$folder_id")
+                    completion="${download_completion}/${upload_completion}"
+                    [ "$download_completion" != "100" ] || [ "$upload_completion" != "100" ] && echo "not_synced" > /tmp/sync_status
+                fi
+
+                status_line="$folder_label:\n$completion%"
+                log_message "  $status_line"
+                summary="${summary}  $status_line\n"
+
+                # Append the status line to the display file with an extra blank line
+                echo "$folder_label:" >> /tmp/sync_display.txt
+                echo "$completion%" >> /tmp/sync_display.txt
+                echo "" >> /tmp/sync_display.txt
+            done
+            summary="${summary}\n"
+            log_message ""
+        done
+
+        # Read the contents of the display file for display
+        status_content=$(cat /tmp/sync_display.txt)
+        display -t "$status_content
+
+Press START to cancel"
+
+        if [ ! -f /tmp/sync_status ]; then
+            log_message "All folders on all devices are in sync."
+            if [ "$mode" != "monitor" ]; then
+                return 0
+            fi
+        fi
+
+        log_message "---"
+        sleep $CHECK_INTERVAL
+    done
+}
+
+set_api_key() {
+    log_message "Reading API key from $CONFIG_XML"
+    if [ ! -f "$CONFIG_XML" ]; then
+        log_message "Error: Config file $CONFIG_XML does not exist" >&2
+        return 1
+    fi
+
+    API_KEY=$(sed -n 's:.*<apikey>\(.*\)</apikey>.*:\1:p' "$CONFIG_XML")
+    log_message "API key: $API_KEY"
+
+    if [ -z "$API_KEY" ]; then
+        log_message "Error: No API key found in config.xml" >&2
+        return 1
+    fi
+}
+
+# Main script
+main() {
+    start_network
+    
+    # Set the API key
+    if ! set_api_key; then
+        display -t "Error: Unable to find API key"
+        sleep 2
+        stop_network
+        exit 1
+    fi
+
+    case "$1" in
+        --monitor)
```

Comment:
This is only used for testing purposes. There were times I just wanted a constant stream of status while SSH'd into the device

Reply:
This would be a good code comment

Reply:
Good call!

cddc54f8e205d76fcc53086559910a5e6a685000